problem_builder/public/js/mentoring_with_steps.js

```diff
@@ -177,6 +177,8 @@ function MentoringWithStepsBlock(runtime, element) {
 
     function updateDisplay() {
         cleanAll();
+        nextDOM.attr('value', $('div.sb-step', element).data('next-button-label'));
```

Comment:
@Kelketek When testing your changes I found that if there are multiple Mentoring Step blocks, setting different "Next step" labels doesn't work the way it's supposed to: The label doesn't change from step to step; it's always set to the `next-button-label` of the first step on the page.

One way of addressing this would be to add a function to `MentoringStepBlock` (in step.js) that returns the label of the current step, something like:

```js
        getStepLabel: function() {
            return $('.sb-step', element).data('next-button-label');
        }
```

... and then update via:

```js
        var step = steps[activeStep];
        nextDOM.attr('value', step.getStepLabel());
```

Also, I think you'll need to update the label not just from `updateDisplay` but also from `jumpToReview` (this is called from `reviewNextStep`, which acts as a click handler for `nextDOM` while reviewing individual steps). To test this (I didn't), set `max_attempts` and `extended_feedback` on the Step Builder block and complete the block in the LMS until you have no attempts left, then use one of the links to jump back to a specific question/step.